plugins/logs/plugin_test.go

```diff
@@ -53,31 +60,170 @@ func TestPluginStart(t *testing.T) {
 
 	chunk1 := <-fixture.server.ch
 	chunk2 := <-fixture.server.ch
-	expLen1 := 153
-	expLen2 := 1
+	chunk3 := <-fixture.server.ch
+	chunk4 := <-fixture.server.ch
+	expLen1 := 144
+	expLen2 := 143
+	expLen3 := 143
+	expLen4 := 70
+
+	if len(chunk1) != expLen1 || len(chunk2) != expLen2 || len((chunk3)) != expLen3 || len(chunk4) != expLen4 {
+		t.Fatalf("Expected chunk lens %v, %v, %v and %v but got: %v, %v, %v and %v", expLen1, expLen2, expLen3, expLen4, len(chunk1), len(chunk2), len(chunk3), len(chunk4))
+	}
+
+	var expInput interface{} = input
+
+	exp := EventV1{
+		Labels: map[string]string{
+			"id":  "test-instance-id",
+			"app": "example-app",
+		},
+		Revision:    "499",
+		DecisionID:  decisionID,
+		Path:        "foo/bar",
+		Input:       &expInput,
+		Result:      &result,
+		RequestedBy: "test",
+		Timestamp:   ts,
+	}
+
+	if !reflect.DeepEqual(chunk4[expLen4-1], exp) {
+		t.Fatalf("Expected %+v but got %+v", exp, chunk4[expLen4-1])
+	}
+}
+
+func TestPluginStartRandomInput2(t *testing.T) {
+	ctx := context.Background()
+
+	fixture := newTestFixture(t)
+	defer fixture.server.stop()
+
+	fixture.server.ch = make(chan []EventV1, 4)
+	var result interface{} = false
 
-	if len(chunk1) != expLen1 || len(chunk2) != expLen2 {
-		t.Fatalf("Expected chunk lens %v and %v but got: %v and %v", expLen1, expLen2, len(chunk1), len(chunk2))
+	var decisionID string
+	ts, err := time.Parse(time.RFC3339Nano, "2018-01-01T12:00:00.123456Z")
+	if err != nil {
+		panic(err)
+	}
+
+	var input map[string]interface{}
+
+	for i := 0; i < 500; i++ {
```

Comment:
Kudos on finding test inputs that generate exactly 500 events!

Reply:
😄